VideoLocker/src/main/java/org/edx/mobile/services/VideoDownloadHelper.java

```diff
@@ -83,26 +87,25 @@ private void startDownloadVideos(List<HasDownloadEntry> model, FragmentActivity
         ArrayList<DownloadEntry> downloadList = new ArrayList<DownloadEntry>();
         int downloadCount = 0;
         for (HasDownloadEntry v : model) {
-            DownloadEntry de = v.getDownloadEntry( storage );
+            DownloadEntry de = v.getDownloadEntry(storage);
             if (de.downloaded == DownloadEntry.DownloadedState.DOWNLOADING
-                || de.downloaded == DownloadEntry.DownloadedState.DOWNLOADED
-                || de.isVideoForWebOnly ) {
+                    || de.downloaded == DownloadEntry.DownloadedState.DOWNLOADED
+                    || de.isVideoForWebOnly) {
                 continue;
             } else {
                 downloadSize = downloadSize
-                    + v.getSize();
+                        + v.getSize();
                 downloadList.add(de);
                 downloadCount++;
             }
         }
         if (downloadSize > MemoryUtil
-            .getAvailableExternalMemory(activity)) {
-            ((CourseDetailTabActivity) activity)
```

Comment:
Bonus! This would have crashed the app if it happened with new course navigation, since the cast would have failed.